src/storage/file/mod.rs

```diff
@@ -329,31 +323,26 @@ impl<T: crate::search::Search + Send + Sync> Storage for FileStorage<T> {
             // not be correct.
             out.flush();
             out.seek(std::io::SeekFrom::Start(0)).await?;
-            trace!("Validating data for SHA {}", sha);
-            validate_sha256(&mut out, label.sha256.as_str()).await?;
-            trace!("SHA {} data validated", sha);
+            trace!("Validating data for SHA {}", parcel_id);
+            validate_sha256(&mut out, parcel_id).await?;
+            trace!("SHA {} data validated", parcel_id);
+            // TODO: Should we also validate length? We use it for returning the proper content length
```

Comment:
I _think_ that we can skip validating content length here because if we know the SHA matches, we know the length must match. Though I guess if the initial length was miscalculated, then this could not match.

Reply:
This is more along making sure the length as indicated by the label matches what was given